### PR TITLE
[#464] Remove placeholder and cancel drop on drag out of bounds

### PR DESCRIPTION
**Pull request by ppuleo**
- Adds the ability to cancel dropping a node in a connected tree configuration when the dragged node is dragged outside of both ui-trees or dragged back to the source ui-tree
- Removes the placeholder when a dragged node is outside of the ui-tree

### CODE REVIEW

**Comment by ppuleo** [comment]
Hey @Voles - no rush on this, it just corrects an issue identified in #464 where a dragged node was always added to a destination tree in a connected tree configuration even when the node was dragged back out of the tree before being dropped.

**Comment by andreaslarssen** [comment]
Nice one! Been waiting for this, thanks!

File: dist/angular-ui-tree.js [hunk 1]
```diff
@@ -771,6 +772,49 @@
...
+
+                outOfBounds = !(targetElm.scope().$type);
+
+                // Detect out of bounds condition, update drop target display, and prevent drop
+                if (outOfBounds) {
+
+                  // Remove the placeholder
+                  placeElm.remove();
+
+                  // If the target was an empty tree, replace the emply element placeholder
```
Comment (Voles) [inline dist/angular-ui-tree.js]: Typo: emply -> empty

**Comment by Voles** [comment]
Another great addition, @ppuleo! I've reviewed your code and found a minor typo (see comment). I'll fix this one myself so this can be merged.

Thanks for your work :cake:

**Pull request merged**